### PR TITLE
enceladus and standardization

### PR DESCRIPTION
**Pull request by AdrianOlosutean**
Initial attept of adding enceladus and standardization code

### CODE REVIEW

File: src/main/scala/za/co/absa/spark/commons/common/Constants.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2018 ABSA Group Limited
```
Comment (Zejnilovic) [inline src/main/scala/za/co/absa/spark/commons/common/Constants.scala:2]: A lot of 2018 years in copyright. Please update to 2020 through the project

Comment (benedeki) [reply src/main/scala/za/co/absa/spark/commons/common/Constants.scala:2]: Why 2020? Enceladus where the files comes from is from 2018. This library on the other hand from 2021,

**Comment by benedeki** [comment]
OK, this is too big and I dare to say, majority of the things does not belong to _spark-commons_.
I would scrap it and add things *class by class*, and starting only with the obvious ones. from my point of view:
1. `SparkTestBase` enhancement
2. `SparkVersionGuard`
3. `DataFrameImplicits`
4. `ColumnImplicits.ColumnEnhancements.isInfinite`, maybe the `zeroBasedSubstr`, the part with `Section` no
5. `StructFieldImplicits`
6. Stuff from `SchemaUtils`, perhaps selective, maybe little refactored (made implicit)
7. Stuff from `SparkUtils`, perhaps selective, maybe little refactored (made implicit)
8. ...
9. Evetually the recordId generation

And also I would very much prefer Hofs and Hats depending on Spark commons than visa versa.

File: src/main/scala/za/co/absa/spark/commons/common/Constants.scala (hunk 1 above)
Comment (Zejnilovic) [reply src/main/scala/za/co/absa/spark/commons/common/Constants.scala]: I am just forgetting what year it is

**Pull request closed without merging**